.github/workflows/code-checker.yml

```diff
@@ -0,0 +1,47 @@
+name: Code Checker
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  linting_and_type-checking:
+    name: Linting, Formatting and Type checking
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - name: Install dependencies
+        run: yarn
+
+      - name: Linting and Formatting checks
+        run: yarn run lint
+
+      - name: Type checking
+        run: yarn run typecheck
+
+  build:
+    name: Build check
+    runs-on: ubuntu-latest
+    needs: linting_and_type-checking 
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - name: Install dependencies
+        run: yarn
```

Comment:
Use  should use `yarn install --frozen-lockfile` instead.

You can check out this npm blog why this is important: [Introducing npm ci for faster, more reliable builds](https://blog.npmjs.org/post/171556855892/introducing-npm-ci-for-faster-more-reliable). 

`yarn install --frozen-lockfile` does the same thing as `npm ci`. Since we are using Yarn, we are good to go.